group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1475,20 +1457,23 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, CoordinatorRecord>
 
         // 2. Update the target assignment if the group epoch is larger than the target assignment epoch or a static member
         // replaces an existing static member. The delta between the existing and the new target assignment is persisted to the partition.
-        int targetAssignmentEpoch = group.assignmentEpoch();
-        Assignment targetAssignment = group.targetAssignment(memberId);
-        if (groupEpoch > targetAssignmentEpoch || staticMemberReplaced) {
+        final int targetAssignmentEpoch;
```

Comment:
nit: should we update the comment if we aren't looking at if a static member is replaced anymore?